src/Lean/MonadEnv.lean

```diff
@@ -118,7 +118,26 @@ def getConstInfoRec [Monad m] [MonadEnv m] [MonadError m] (constName : Name) : m
   | ConstantInfo.recInfo v => pure v
   | _                      => throwError "'{mkConst constName}' is not a recursor"
 
-@[inline] def matchConstStruct [Monad m] [MonadEnv m] [MonadError m] (e : Expr) (failK : Unit → m α) (k : InductiveVal → List Level → ConstructorVal → m α) : m α :=
+/--
+Matches if `e` is a constant that is an inductive type with one constructor.
+Such types can be used with primitive projections.
+See also `Lean.matchConstStructLike` for a more restrictive version.
+-/
+@[inline] def matchConstStructure [Monad m] [MonadEnv m] [MonadError m] (e : Expr) (failK : Unit → m α) (k : InductiveVal → List Level → ConstructorVal → m α) : m α :=
```

Comment:
Should we rename these functions in the future? It is ok to merge with the current names, but I guess we will get confused in the future. I am referring to `matchConstStructure` and `matchConstStructureLike`

Reply:
Suggestion: `matchConstStructure` and `matchConstNonRecStructure`

Reply:
Created an issue to track this suggestion #5891